repo2docker/buildpacks/pipfile/__init__.py

```diff
@@ -74,6 +75,43 @@ def python_version(self):
             self._python_version = self.major_pythons["3"]
             return self._python_version
 
+    @property
+    def _should_preassemble_pipenv(self):
+        """Peek in Pipfile.lock to determine if we can assemble from only env files
+
+        If there are any local references, e.g. `"path": "./package.tar.gz"`,
+        stage the whole repo prior to installation.
+        """
+        if not os.path.exists("binder") and os.path.exists("setup.py"):
```

Comment:
I think this should use https://github.com/jupyter/repo2docker/blob/e6cbee56a8852674c817b1b2f17d040f4805f6ef/repo2docker/buildpacks/base.py#L503 to take care of `binder/` and `.binder/` (and what ever other sub-directories we might add in the future)